uvicorn/config.py

```diff
@@ -122,6 +124,8 @@ def __init__(
         self.ssl_cert_reqs = ssl_cert_reqs
         self.ssl_ca_certs = ssl_ca_certs
         self.ssl_ciphers = ssl_ciphers
+        self._headers = headers if headers else []
+        self.headers = None  # type: List[Tuple[bytes, bytes]]
```

Comment:
Thinking about this once more - How about we go for `.headers` as the string pairs, and `.encoded_headers` for the byte-encoded pairs?